components/lib/dataprotect/src/main/java/mozilla/components/lib/dataprotect/SecureAbove22Preferences.kt

```diff
@@ -82,13 +85,36 @@ class SecureAbove22Preferences(context: Context, name: String) : KeyValuePrefere
  * A simple [KeyValuePreferences] implementation which entirely delegates to [SharedPreferences] and doesn't perform any
  * encryption/decryption.
  */
-private class InsecurePreferencesImpl21(context: Context, name: String) : KeyValuePreferences {
+private class InsecurePreferencesImpl21(
+    context: Context,
+    name: String,
+    migrateFromSecureStorage: Boolean = true
+) : KeyValuePreferences {
     companion object {
         private const val SUFFIX = "_kp_pre_m"
     }
 
     private val prefs = context.getSharedPreferences("$name$SUFFIX", MODE_PRIVATE)
 
+    init {
+        // Check if we have any encrypted values stored on disk.
+        if (migrateFromSecureStorage && Build.VERSION.SDK_INT >= M) {
+            if (prefs.all.isEmpty()) {
+                val secureStorage = SecurePreferencesImpl23(context, name, false)
+                // Copy over any old values.
+                try {
+                    secureStorage.all().forEach {
+                        putString(it.key, it.value)
+                    }
+                } catch (e: Error) {
```

Comment:
You don't do the try/catch in the other storage (since we don't believe it can fail). Can you add a comment here, describing why this is necessary?